git-publish

```diff
@@ -837,18 +837,15 @@ branch.%s.pushRemote is set appropriately?  (Override with --no-url-check)''' %
                              extra_args=args)
             if message:
                 cover_letter_path = os.path.join(tmpdir, '0000-cover-letter.patch')
-                msg = email.message_from_binary_file(open(cover_letter_path, 'rb'),
-                                                     policy=email.policy.SMTP)
 
-                subject = msg['Subject'].replace('\n', '')
-                subject = subject.replace('*** SUBJECT HERE ***', message[0])
-                msg.replace_header('Subject', subject)
+                cover_letter_data = open(cover_letter_path, "rt").read()
```

Comment:
The cover letter is generated by git-format-patch(1), so I'm not sure it follows the same rules.

Anyway, should I use the locale preferred encoding like in the following patch?

```diff
diff --git a/git-publish b/git-publish
index 92fcacb..0784af3 100755
--- a/git-publish
+++ b/git-publish
@@ -838,14 +838,14 @@ branch.%s.pushRemote is set appropriately?  (Override with --no-url-check)''' %
             if message:
                 cover_letter_path = os.path.join(tmpdir, '0000-cover-letter.patch')
 
-                cover_letter_data = open(cover_letter_path, "rt").read()
+                cover_letter_data = open(cover_letter_path, "rb").read().decode(ENCODING)
 
                 cover_letter_data = re.sub('\*\*\*\s+SUBJECT\s*HERE\s+\*\*\*', message[0], cover_letter_data)
 
                 blurb = os.linesep.join(message[2:])
                 cover_letter_data = cover_letter_data.replace('*** BLURB HERE ***', blurb)
 
-                open(cover_letter_path, 'wt').write(cover_letter_data)
+                open(cover_letter_path, 'wb').write(cover_letter_data.encode(ENCODING))
             patches = sorted(glob.glob(os.path.join(tmpdir, '*')))
             if options.annotate:
                 edit(*patches)

```

Reply:
I don't know the answer. It depends on whether git-format-patch(1) encodes the output in a certain way or just uses the system encoding. I think it will be necessary to try it out or read the git source, sorry!

Reply:
Thanks, I'll take a look at the code.

Reply:
I took a quick look at git source code (`git-format-patch` is implemented in builtin/log.c).
[The encoding of the cover letter is set to UTF-8](https://github.com/git/git/blob/54e85e7af1ac9e9a92888060d6811ae767fea1bc/builtin/log.c#L1157) (and does not seem to be configurable).

If there are non-ASCII characters, the CTE is set to 8bit (and does not seem to be configurable). CTE is also added in the  header in [pp_titile_line()](https://github.com/git/git/blob/54e85e7af1ac9e9a92888060d6811ae767fea1bc/pretty.c#L1795) called in [prepare_cover_text()](https://github.com/git/git/blob/54e85e7af1ac9e9a92888060d6811ae767fea1bc/builtin/log.c#L1121).

That said, maybe is better to revert 0f2dd0b ("Avoid to encode the cover letter") and force the encoding to UTF-8 with CTE=8bit, putting a big comment with these info.

I'll check better if it works well when `sendemail.transferEncoding` is set to a different value.

Reply:
@stefano-garzarella Did you find out how `sendemail.transferEncoding` works?

Reply:
@stefanha not yet, but I will look at it today ;-)

Reply:
@stefanha forcing UTF-8 with CTE=8bit on the cover letter works well with `git-send-email(1)`, also if `sendemail.transferEncoding` is set to a different value (I tried auto, 8bit, quoted-printable, and base64).

I'll revert 0f2dd0bd9e and push the patch.

I'll also try to fix #96